js/windowMgr.js

```diff
@@ -500,6 +504,77 @@ function doCreateMainWindow(initialUrl, initialBounds, isCustomTitleBar) {
+    
+    // ELECTRON-323: Handle session permission requests
+    function handlePermissionRequests(webContents) {
+        
+        let session = webContents.session;
+        
+        getConfigField('permissions')
+            .then((permissions) => {
+                
+                if (!permissions) {
+                    log.send(logLevels.ERROR, 'permissions configuration is invalid, so, everything will be true by default!');
+                    return;
+                }
+                
+                session.setPermissionRequestHandler((sessionWebContents, permission, callback) => {
+                    
+                    function handleSessionPermissions(userPermission, message, cb) {
+                        
+                        log.send(logLevels.INFO, 'permission is -> ' + userPermission);
+                        
+                        if (!userPermission) {
+                            let fullMessage = `Your administrator has disabled ${message}. Please contact your admin for help.`;
+                            electron.dialog.showErrorBox('Permission Denied!', fullMessage);
+                        }
+                        
+                        return cb(userPermission);
+                        
+                    }
+                    
+                    let PERMISSION_MEDIA = 'media';
+                    let PERMISSION_LOCATION = 'geolocation';
+                    let PERMISSION_NOTIFICATIONS = 'notifications';
+                    let PERMISSION_MIDI_SYSEX = 'midiSysex';
+                    let PERMISSION_POINTER_LOCK = 'pointerLock';
+                    let PERMISSION_FULL_SCREEN = 'fullscreen';
+                    let PERMISSION_OPEN_EXTERNAL = 'openExternal';
+                    
+                    switch (permission) {
+                        
+                        case PERMISSION_MEDIA:
+                            return handleSessionPermissions(permissions.media, 'sharing your camera, microphone, and speakers', callback);
+                        
+                        case PERMISSION_LOCATION:
+                            return handleSessionPermissions(permissions.geolocation, 'sharing your location', callback);
+                        
+                        case PERMISSION_NOTIFICATIONS:
+                            return handleSessionPermissions(permissions.notifications, 'notifications', callback);
+                        
+                        case PERMISSION_MIDI_SYSEX:
+                            return handleSessionPermissions(permissions.midiSysex, 'MIDI Sysex', callback);
+                        
+                        case PERMISSION_POINTER_LOCK:
+                            return handleSessionPermissions(permissions.pointerLock, 'Pointer Lock', callback);
+                        
+                        case PERMISSION_FULL_SCREEN:
+                            return handleSessionPermissions(permissions.fullscreen, 'Full Screen', callback);
```

Comment:
web-rtc based meetings support full screen - doesn't this mean this button won't work in our UI???

Reply:
it can if the appropriate permissions are provided by the admin. if the admin has specifically disabled full screen, it won't work in that case.

Reply:
when disabled, will the button still show as enabled for web-rtc?  maybe web-rtc needs to check if full screen permissions are allowed and remove/disable the button.  if applies, then should create a ticket for them.

Reply:
yes, the button is displayed, but if the button is clicked, we show an error dialog in electron saying "Permission Denied" with a detailed message that full screen has been disabled by the admin.

and sure, I'll talk to the rtc team as well. actually, we'll also need to check with the design team to get approval on the UX.

please note that this is at a nascent stage and will be built upon in the future versions.